src/test/regress/expected/failure_ddl.out

```diff
@@ -0,0 +1,1228 @@
+
+-- killing on PREPARE should be fine, everything should be rollbacked
+SELECT citus.mitmproxy('conn.onCommandComplete(command="PREPARE TRANSACTION").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+ALTER TABLE test_table DROP COLUMN new_column;
+ERROR:  connection not open
+CONTEXT:  while executing command on localhost:57640
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM public.table_attrs where relid = 'test_table'::regclass;
+ count 
+-------
+     3
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100800,t,3)
+ (localhost,57637,100802,t,3)
+ (localhost,57640,100801,t,3)
+ (localhost,57640,100803,t,3)
+(4 rows)
+
+-- we should be able to revocer the transaction and
+-- see that the command is rollbacked
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions 
+-------------------------------
+                             2
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100800,t,3)
+ (localhost,57637,100802,t,3)
+ (localhost,57640,100801,t,3)
+ (localhost,57640,100803,t,3)
+(4 rows)
+
+-- cancelling on PREPARE should be fine, everything should be rollbacked
+SELECT citus.mitmproxy('conn.onCommandComplete(command="PREPARE TRANSACTION").cancel(' ||  pg_backend_pid() || ')');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+ALTER TABLE test_table DROP COLUMN new_column;
+ERROR:  canceling statement due to user request
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM public.table_attrs where relid = 'test_table'::regclass;
+ count 
+-------
+     3
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100800,t,3)
+ (localhost,57637,100802,t,3)
+ (localhost,57640,100801,t,3)
+ (localhost,57640,100803,t,3)
+(4 rows)
+
+-- we should be able to revocer the transaction and
+-- see that the command is rollbacked
+SELECT recover_prepared_transactions();
+ recover_prepared_transactions 
+-------------------------------
+                             1
+(1 row)
+
+SELECT run_command_on_placements('test_table', $$SELECT count(*) FROM public.table_attrs where relid = '%s'::regclass$$) ORDER BY 1;
+  run_command_on_placements   
+------------------------------
+ (localhost,57637,100800,t,3)
+ (localhost,57637,100802,t,3)
+ (localhost,57640,100801,t,3)
+ (localhost,57640,100803,t,3)
+(4 rows)
+
+-- killing on command complete of COMMIT PREPARE, we should see that the command succeeds
+-- and all the workers committed
```

Comment:
This comment is a little confusing, I wish we could show the `WARNING`s here. From the coordinator's perspective the query "succeeds" in the exact same way that it does below. We give a warning, and can't be sure whether the changes have actually been applied until you run `recover_prepared_transactions()`.

Reply:
Opened an issue: https://github.com/citusdata/citus/issues/2282